tests/unit/contracts/test_forbidden.py

```diff
@@ -138,6 +138,15 @@ def test_ignore_imports_tolerates_duplicates(self):
         )
         assert contract.check(graph=graph)
 
+    def test_allow_indirect_imports(self):
+        graph = self._build_graph()
+        contract = self._build_contract(
+            forbidden_modules=("mypackage.purple"),
+            allow_indirect_imports=True,
```

Comment:
In reality `allow_indirect_imports` will be a string, not a boolean.

For completeness, I suggest parametrizing this test so we check different values and the result, along the lines of:

```
    @pytest.mark.parametrize(
        "allow_indirect_imports, contract_is_kept",
        ((None, False), ("false", False), ("True", True), ("true", True), ("anything", False)),
    )
```